aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4SigningDecorator.kt

```diff
@@ -55,8 +56,10 @@ class SigV4SigningDecorator : RustCodegenDecorator {
         return baseCustomizations.letIf(applies(protocolConfig)) { customizations ->
             val serviceHasEventStream = protocolConfig.serviceShape.operations
                 .any { id ->
-                    protocolConfig.model.expectShape(id, OperationShape::class.java)
-                        .isEventStream(protocolConfig.model)
+                    // Some models like `kinesisanalytics` have StructureShapes in their operation list,
+                    // so don't assume they're always OperationShape.
+                    val shape = protocolConfig.model.getShape(id).orNull()
+                    shape is OperationShape && shape.isEventStream(protocolConfig.model)
```

Comment:
might want to extract this as an extension function on ServiceShape if it's needed elsewhere